hibernate-core/src/main/java/org/hibernate/engine/jdbc/ResultSetWrapperProxy.java

```diff
@@ -96,20 +119,19 @@ private Integer findColumn(String columnName) throws SQLException {
 		return columnNameCache.getIndexForColumnName( columnName, rs );
 	}
 
-	private boolean isFirstArgColumnLabel(Method method, Object[] args) {
+	private static boolean isFirstArgColumnLabel(Method method) {
 		// method name should start with either get or update
 		if ( ! ( method.getName().startsWith( "get" ) || method.getName().startsWith( "update" ) ) ) {
 			return false;
 		}
 
 		// method should have arguments, and have same number as incoming arguments
-		if ( ! ( method.getParameterCount() > 0 && args.length == method.getParameterCount() ) ) {
```

Comment:
Not sure why you removed that. If you're trying to fix a bug while you're at it, maybe a separate commit would be better. But I wouldn't dare do that unless I had a very specific reason: I'm not sure there are lots of tests for this class.

Also you should update the comment just above.

Reply:
It's related as I don't have the `args` parameter anymore when taking this decision: it's done only based on the method.

I will update the comment.

Reply:
Got it, the `args` parameter is replaced by the `parameterTypes` member in 
 `ResultSetMethodKey`.